subworkflows/local/bam_applybqsr_spark/main.nf

```diff
@@ -18,23 +18,12 @@ workflow BAM_APPLYBQSR_SPARK {
     main:
     ch_versions = Channel.empty()
 
-    cram_intervals = cram.combine(intervals)
-        .map{ meta, cram, crai, recal, intervals, num_intervals ->
-
-            //If no interval file provided (0) then add empty list
-            intervals_new = num_intervals == 0 ? [] : intervals
-
-            [[
-                id:             meta.sample,
-                data_type:      meta.data_type,
-                num_intervals:  num_intervals,
-                patient:        meta.patient,
-                sample:         meta.sample,
-                sex:            meta.sex,
-                status:         meta.status,
-            ],
-            cram, crai, recal, intervals_new]
-        }
+    cram_intervals = cram.combine(intervals).map{ meta, cram, crai, recal, intervals, num_intervals ->
+        //If no interval file provided (0) then add empty list
+        [ meta.subMap('data_type', 'patient', 'sample', 'sex', 'status')
+            + [num_intervals:num_intervals, id: meta.sample],
```

Comment:
I'm concerned here that id is now meta.sample, aren't we loosing some info there?

Reply:
:facepalm: